pkg/controller/controller.go

```diff
@@ -620,6 +620,19 @@ func (c *controller) updateInstanceFinalizers(
 	instance *v1alpha1.Instance,
 	finalizers []string) error {
 
+	// Get the latest version of the instance so that we can avoid conflicts
+	// (since we have probably just updated the status of the instance and are
+	// now removing the last finalizer).
+	//
+	// It's important to do everything we can to ensure that we complete this
```

Comment:
It means that if you do run into a conflict here, and try to re-deprovision or re-unbind, a misbehaved broker might not correctly return a 410 and lead you to believe there's some kind of error (by giving you a 500, for example).